lnwallet/rpcwallet/remote_signer.go

```diff
@@ -86,3 +93,167 @@ type RemoteSignerRequests interface {
+// watch-only node to connect to it via an inbound GRPC connection.
+type InboundRemoteSigner struct {
+	// Embedded signrpc.SignerClient and walletrpc.WalletKitClient to
+	// implement the RemoteSigner interface.
+	signrpc.SignerClient
+
+	walletrpc.WalletKitClient
+
+	// The host:port of the remote signer node.
+	rpcHost string
+
+	// The path to the TLS certificate of the remote signer node.
+	tlsCertPath string
+
+	// The path to the macaroon of the remote signer node.
+	macaroonPath string
+
+	// The timeout for the connection to the remote signer node.
+	timeout time.Duration
+}
+
+// NewInboundRemoteSigner creates a new InboundRemoteSigner instance.
+// The function sets up a connection to the remote signer node.
+// The returned function is a cleanup function that should be called to close
+// the connection when the remote signer is no longer needed.
+func NewInboundRemoteSigner(rpcHost string, tlsCertPath string,
+	macaroonPath string,
+	timeout time.Duration) (*InboundRemoteSigner, func(), error) {
+
+	rpcConn, err := connect(rpcHost, tlsCertPath, macaroonPath, timeout)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error connecting to the remote "+
+			"signing node through RPC: %v", err)
+	}
+
+	cleanUp := func() {
+		rpcConn.Close()
+	}
+
+	remoteSigner := &InboundRemoteSigner{
+		SignerClient:    signrpc.NewSignerClient(rpcConn),
+		WalletKitClient: walletrpc.NewWalletKitClient(rpcConn),
+		rpcHost:         rpcHost,
+		tlsCertPath:     tlsCertPath,
+		macaroonPath:    macaroonPath,
+		timeout:         timeout,
+	}
+
+	return remoteSigner, cleanUp, nil
+}
+
+// Run feeds lnd with the incoming stream set up by an outbound remote signer
+// and then blocks until the stream is closed. Lnd can then send any requests to
+// the remote signer through the stream.
+//
+// NOTE: This is part of the RemoteSigner interface.
+func (*InboundRemoteSigner) Run(
+	_ walletrpc.WalletKit_SignCoordinatorStreamsServer) error {
+
+	// If lnd has been configured to use an inbound remote signer, it should
+	// not allow an outbound remote signer to connect.
+	return errors.New("not supported when remotesigner.signertype is set " +
+		"to \"inbound\"")
+}
+
+// Ready blocks and returns nil when the remote signer is ready to accept
+// requests.
+//
+// NOTE: This is part of the RemoteSigner interface.
+func (r *InboundRemoteSigner) Ready() error {
+	// The inbound remote signer is ready as soon we have connected to the
+	// remote signer node in the constructor. Therefore, we always return
+	// nil here to signal that we are ready.
+	return nil
+}
+
+// Ping verifies that the remote signer is still responsive.
+//
+// NOTE: This is part of the RemoteSigner interface.
+func (r *InboundRemoteSigner) Ping(timeout time.Duration) error {
+	conn, err := connect(
+		r.rpcHost, r.tlsCertPath, r.macaroonPath,
+		timeout,
+	)
+	if err != nil {
+		return fmt.Errorf("error connecting to the remote "+
+			"signing node through RPC: %v", err)
+	}
+
+	defer func() {
```

Comment:
Why is this in a `defer`? Since we're done now?
And not sure if we also want to report errors when not being able to disconnect?
So we might just be able to do `return conn.Close()` here instead?